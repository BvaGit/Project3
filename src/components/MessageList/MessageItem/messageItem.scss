@import "../../../utils/variables.scss";

.message-item {
  font-family: "Noto Sans KR", sans-serif;
  position: relative;
  background: #f2f2f2;
  width: 700px;
  min-height: 70px;
  margin-bottom: $margin_lg;
  border-radius: 10px;
  box-shadow: 0px 10px 13px -7px #000000, 5px 5px 15px 5px rgba(0, 0, 0, 0.09);
  color: $title-color;
  padding: 10px 15px 30px;
  overflow-wrap: break-word;

  &--user {
    margin-left: auto;
  }

  &--member {
    margin-right: auto;
  }

  &__block {
    display: flex;
    justify-content: space-between;
    align-items: center;
    overflow: hidden;
  }

  &__name {
    position: absolute;
    right: 25px;
    top: -10px;
    font-size: 12px;
    color: #fff;
    background-color: #209a9f;
    padding: 2px;
  }

  &__date {
    text-align: right;
    position: absolute;
    bottom: 10px;
    right: 15px;
    font-size: 12px;
  }
}

.theme-dark {
  .message-item {
    background: #292929;
    color: $l-body-text-color;
  }
}

.avatar {
  display: block;
  width: 50px;
  height: 50px;
  border-radius: 50%;
  overflow: hidden;
  img {
    width: 100%;
  }
}
