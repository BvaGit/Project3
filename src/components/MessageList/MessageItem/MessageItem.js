import React from "react";

import moment from "moment-timezone";

import "./messageItem.scss";

const MessageItem = ({ messages = [], userId, name, lastMessageBlock }) => {
  return (
    <div className='message-box__container'>
      {messages.map((message, index) => (
        <div
          ref={messages.length - 1 === index ? lastMessageBlock : null}
          key={message.message_id}
          className={
            +userId === message.id
              ? "message-item message-item--user"
              : "message-item message-item--member"
          }
        >
          <div className="message-item__block">
            {message.content}
            <p className="message-item__name"> {message.login || name}</p>
            <span className="avatar">
              <img
                src={
                  message.avatar
                    ? message.avatar
                    : "../../../../public/assets/images/1-36-1024x1024.jpg"
                }
              />
            </span>
          </div>
          <span className="message-item__date">
            {moment(message.date_create).format("LT")}
          </span>
        </div>
      ))}
    </div>
  );
};

export default MessageItem;
