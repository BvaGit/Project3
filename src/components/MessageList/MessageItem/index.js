import { connect } from "react-redux";

import MessageItem from "./MessageItem";

import { getUserID } from "../../../store/user/selectors";

const mapStateToProps = (state) => ({
  userId: getUserID(state),
});

export default connect(mapStateToProps)(MessageItem);
