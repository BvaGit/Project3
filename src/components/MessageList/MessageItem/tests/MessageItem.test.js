import React from 'react';
import { shallow, mount } from 'enzyme';
import MessageItem from '../../MessageItem/MessageItem';



describe('MessageItem', () => {
    it("Should match snapshot", () => {
        const messages=[{message_id:1, id:1, content:'content', date_create:'12.34'}]
        const userId="1"
        const component = shallow(<MessageItem messages={messages} userId={userId}/>)
        expect(component.html()).toMatchSnapshot()
    })

}) 