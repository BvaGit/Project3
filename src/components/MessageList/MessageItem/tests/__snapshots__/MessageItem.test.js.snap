// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem Should match snapshot 1`] = `"<div class=\\"message-item message-item--user\\">content<span class=\\"message-item__date\\">Invalid date</span></div>"`;
